src/SymEngine.jl

```diff
@@ -149,3 +149,14 @@ show(io::IO, b::Basic) = print(io, toString(b))
 
 end
 
+abstract Basic
+
```

Comment:
it is not possible.  It cannot be both abstract and concrete type . @isuruf
So we have to build  hierarchy only with abstract types ?